gdk4/src/rectangle.rs

```diff
@@ -59,3 +75,20 @@ impl From<RectangleInt> for Rectangle {
         unsafe { *(&r as *const _ as *const _) }
     }
 }
+
+impl From<cairo::Rectangle> for Rectangle {
+    // rustdoc-stripper-ignore-next
+    /// Note that converting between a [`cairo::Rectangle`] and [`Rectangle`]
+    /// will probably lead to precisison errors. Use cautiously.
+    fn from(r: cairo::Rectangle) -> Self {
+        skip_assert_initialized!();
+        Self::new(r.x as i32, r.y as i32, r.width as i32, r.height as i32)
+    }
+}
+
+impl From<pango::Rectangle> for Rectangle {
```

Comment:
Add both also in the other direction `impl From<Rectangle> for pango::Rectangle { ... }` etc